pkg/kritis/metadata/metadata_test.go

```diff
@@ -0,0 +1,263 @@
+)
+
+func TestGetVulnerabilityFromOccurence(t *testing.T) {
+	tests := []struct {
+		name        string
+		severity    vulnerability.Severity
+		fixKind     pkg.Version_VersionKind
+		noteName    string
+		expectedVul Vulnerability
+	}{
+		{"fix available", vulnerability.Severity_LOW,
+			pkg.Version_MAXIMUM,
+			"CVE-1",
+			Vulnerability{
+				CVE:             "CVE-1",
+				Severity:        "LOW",
+				HasFixAvailable: false,
+			},
+		},
+		{"fix not available", vulnerability.Severity_MEDIUM,
+			pkg.Version_NORMAL,
+			"CVE-2",
+			Vulnerability{
+				CVE:             "CVE-2",
+				Severity:        "MEDIUM",
+				HasFixAvailable: true,
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			vulnDetails := &grafeas.Occurrence_Vulnerability{
+				Vulnerability: &vulnerability.Details{
+					Severity: tc.severity,
+					PackageIssue: []*vulnerability.PackageIssue{
+						{
+							AffectedLocation: &vulnerability.VulnerabilityLocation{},
+							FixedLocation: &vulnerability.VulnerabilityLocation{
+								Version: &pkg.Version{
+									Kind: tc.fixKind,
+								},
+							},
+						},
+					},
+				}}
+			occ := &grafeas.Occurrence{
+				NoteName: tc.noteName,
+				Details:  vulnDetails,
+			}
+
+			actualVuln := GetVulnerabilityFromOccurrence(occ)
+			if !reflect.DeepEqual(*actualVuln, tc.expectedVul) {
+				t.Fatalf("Expected \n%v\nGot \n%v", tc.expectedVul, actualVuln)
+			}
+		})
+	}
+}
+
+func TestGetRawAttestationsFromOccurrence(t *testing.T) {
+	tests := []struct {
+		name            string
+		noteName        string
+		att             attestation.Attestation
+		expectedRawAtts []RawAttestation
+	}{
+		{
+			"pgp attestation",
+			"CVE-1",
+			makeOccAttestationPgp("sig-1", "id-1"),
+			[]RawAttestation{
+				makeRawAttestationPgp("sig-1", "id-1"),
+			},
+		},
+		{
+			"generic attestation",
+			"CVE-2",
+			makeOccAttestationGeneric([]string{"sig-1"}, []string{"id-1"}, "generic-address"),
+			[]RawAttestation{
+				makeRawAttestationGeneric("sig-1", "id-1", "generic-address"),
+			},
+		},
+		{
+			"generic attestation multiple signatures",
+			"CVE-3",
+			makeOccAttestationGeneric([]string{"sig-1", "sig-2"}, []string{"id-1", "id-2"}, "generic-address"),
+			[]RawAttestation{
+				makeRawAttestationGeneric("sig-1", "id-1", "generic-address"),
+				makeRawAttestationGeneric("sig-2", "id-2", "generic-address"),
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			occ := &grafeas.Occurrence{
+				NoteName: tc.noteName,
+				Details: &grafeas.Occurrence_Attestation{
+					Attestation: &attestation.Details{Attestation: &tc.att},
+				},
+			}
+			actualRawAtts, err := GetRawAttestationsFromOccurrence(occ)
+			if err != nil {
+				t.Fatalf("Error while parsing RawAttestation from Occurrence: %v", err)
+			}
+			if !cmp.Equal(actualRawAtts, tc.expectedRawAtts, cmpopts.SortMaps(strCompare)) {
```

Comment:
just to confirm since idk: does a custom string compare function really needed here?

Reply:
We discussed offline, we'll use a custom compare function that operates on the RawAttestation's directly.